src/isomorphic/classic/types/ReactPropTypes.js

```diff
@@ -235,19 +235,38 @@ function createEnumTypeChecker(expectedValues) {
 
   function validate(props, propName, componentName, location, propFullName) {
     var propValue = props[propName];
+    var isOneOfPropType = false;
     for (var i = 0; i < expectedValues.length; i++) {
       if (is(propValue, expectedValues[i])) {
         return null;
       }
+      if (getPropType(expectedValues[i]) === 'function' && expectedValues[i].name === 'bound checkType') {
```

Comment:
Could you set a flag on the bound `checkType` instead? This way if we change its name, this won’t start creating false positives.